simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        numpy.array
+            The edges of the direction histograms in y
+        numpy.ndarray
+            The values (counts) of the histogram.
+
+        Raises
+        ------
+        TypeError:
+            if density is not of type bool.
+        HistogramNotCreated:
+            if the histogram was not previously created.
+        """
+        if density is True:
+            return self._get_2D("density")
+        else:
+            return self._get_2D("counts")
+
+    def get_2D_photon_direction_distr(self):
+        """
+        Get 2D histograms of incoming direction of the Cherenkov photons on the ground.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in cos(x).
+        numpy.array
+            The edges of the direction histograms in cos(y)
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_2D("direction")
+
+    def get_2D_photon_time_altitude(self):
+        """
+        Get 2D histograms of the time and altitude of the photon production.
+
+        Returns
+        -------
+        numpy.array
+            The edges of the direction histograms in ns.
+        numpy.array
+            The edges of the direction histograms in km.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        return self._get_2D("time_altitude")
+
+    def get_2D_num_photons_distr(self):
+        """
+        Get the distribution of Cherenkov photons per event per telescope.
+
+        Parameters
+        ----------
+        bins: float
+            Number of bins for the histogram.
+        range: 2-tuple
+            Tuple to define the range of the histogram.
+
+        Returns
+        -------
+        numpy.array
+            Number of photons per event per telescope.
+        numpy.ndarray
+            The values (counts) of the histogram.
+        """
+        num_of_photons_per_event_per_telescope = np.array(
+            self.get_num_photons_per_event_per_telescope()
+        )
+        num_events_array = np.arange(self.num_events + 1)
+        telescope_indices_array = np.arange(self.num_telescopes + 1)
+        return num_events_array, telescope_indices_array, num_of_photons_per_event_per_telescope
+
+    def _get_1D(self, label):
+        """
+        Helper function to get 1D distributions.
+
+        Parameters
+        ----------
+        label: str
+            Label to indicate which histogram.
+
+        Raises
+        ------
+        ValueError:
+            if label is not valid.
+        """
+
+        if label not in self._allowed_1D_labels:
+            msg = "`label` is not valid. Valid entries are {}".format(self._allowed_1D_labels)
+            self._logger.error(msg)
+            raise ValueError
+        self._raise_if_no_histogram()
+
+        x_edges_list, hist_1D_list = [], []
+        for step, _ in enumerate(self.hist_position):
```

Comment:
I did change `step` to `one_hist` as before but I don't understand what do you mean with the `enumerate`

Reply:
Forget the `enumerate` (forgot `hist_position` is not an integer I think). However, as above, `one_hist` should be `i_hist` because it's an integer iterator rather than one histogram position instance.